### PR TITLE
Feature/federation submission payload

### PR DESCRIPTION
**Pull request by mibrasap**

### CODE REVIEW

File: common/persistence/src/main/java/app/coronawarn/server/common/persistence/domain/DiagnosisKeyBuilders.java [hunk 1]
```diff
@@ -45,7 +45,16 @@
      * @param protoBufObject ProtocolBuffer object associated with the temporary exposure key.
      * @return this Builder instance.
      */
-    FinalBuilder fromProtoBuf(TemporaryExposureKey protoBufObject);
+    FinalBuilder fromTemporaryExposureKey(TemporaryExposureKey protoBufObject);
+
+    /**
+     * Adds the data contained in the specified federation diagnosis key key object to this builder.
```
Comment (ioangut) [inline common/persistence/src/main/java/app/coronawarn/server/common/persistence/domain/DiagnosisKeyBuilders.java:51]: Duplicated word "key"

File: common/persistence/src/main/java/app/coronawarn/server/common/persistence/domain/DiagnosisKeyBuilders.java [hunk 2]
```diff
@@ -45,7 +45,16 @@
...
+     *
+     * @param federationDiagnosisKey DiagnosisKey object associated with the temporary exposure key.
+     * @return this Builder instance.
+     */
+    FinalBuilder fromFederationDiagnosisKey(
+        app.coronawarn.server.common.protocols.external.exposurenotification.DiagnosisKey federationDiagnosisKey);
```
Comment (hilmarf) [inline common/persistence/src/main/java/app/coronawarn/server/common/persistence/domain/DiagnosisKeyBuilders.java:57]: @mibrasap please add 'app.coronawarn.server.common.protocols.external.exposurenotification.DiagnosisKey' to the import and use here the short name then

File: common/protocols/src/main/proto/app/coronawarn/server/common/protocols/external/exposurenotification/diagnosis_key_batch.proto [hunk 3]
```diff
@@ -6,5 +6,20 @@ option java_multiple_files = true;
...
+  uint32 rollingPeriod = 3; // number of 10-minute windows between key rolling
+  int32 transmissionRiskLevel = 4; // risk of transmission
+  repeated string visitedCountries = 5;
+  string origin = 6; // country of origin
+  VerificationType verificationType = 7; // set by backend
+}
+
+enum VerificationType {
+  LAB_VERIFIED = 0;
+  SELF_REPORTED = 1;
```
Comment (hilmarf) [inline common/protocols/src/main/proto/app/coronawarn/server/common/protocols/external/exposurenotification/diagnosis_key_batch.proto:24]: any ideas how to keep this in sync with https://github.com/eu-federation-gateway-service/efgs-federation-gateway/blob/master/src/main/proto/Efgs.proto ?

File: common/persistence/src/main/resources/db/migration/V5__updateDiagnosisKeyTableForFederation.sql [hunk 4]
```diff
@@ -1,5 +1,8 @@
 -- Migrate current values -> origin_country = DE
 
+ALTER TABLE diagnosis_key
+    ADD consent_to_federation BOOLEAN NOT NULL DEFAULT FALSE;
```
Comment (hilmarf) [inline common/persistence/src/main/resources/db/migration/V5__updateDiagnosisKeyTableForFederation.sql:4]: where do we store for which countries a key is valid? I'm not 100% sure, but maybe the customer want to ask people to give their consent per country for their keys...

**Pull request merged**